alphagov/smart-answers
# Split marriage-abroad outcome for opposite sex ceremonies in affirmation countries

This branch extracts 15 country-specific outcome templates from the rather complicated outcome_opposite_sex_marriage_in_affirmation_countries template. I think this probably results in a small amount of duplicated content (that isn't already in partials) between the new templates. I'm hoping that we can gradually remove that over time.

The following list (corresponding to the `OS_AFFIRMATION_COUNTRIES` array) contains the 20 countries that previously lead to the outcome_opposite_sex_marriage_in_affirmation_countries template. I've indicated (with `(*)`) the countries that now have their own outcomes:

* Belgium (*)
* Cambodia
* Colombia
* China (*)
* Ecuador (*)
* Egypt (*)
* Lebanon
* Finland
* Macao (*)
* Mongolia
* Morocco (*)
* Norway (*)
* Peru (*)
* Philippines (*)
* Qatar (*)
* South Korea (*)
* Thailand (*)
* Turkey (*)
* United Arab Emirates (*)
* Vietnam (*)

There are quite a few commits but I don't think they should be _too_ hard to follow.

The first two commits inline two partials to the outcome_opposite_sex_marriage_in_affirmation_countries template in preparation for extracting country-specific outcomes.

There are then 4 commits for each of the 15 countries:

1. Add separate condition for the country to marriage-abroad flow

2. Add new outcome template for the country. The template is a duplicate outcome_opposite_sex_marriage_in_affirmation_countries.

3. Remove country-specific logic and content from outcome_opposite_sex_marriage_in_affirmation_countries

4. Remove all logic and content not related to the country from the outcome template created in step 2.

I generated regression test artefacts for all countries before making this change and used those to double check that I haven't altered the behaviour of the system.
